src/rssitem.cpp

```diff
@@ -262,11 +262,16 @@ void RssItem::set_feedptr(const std::weak_ptr<RssFeed>& ptr)
 	feedptr_ = ptr;
 }
 
-void RssItem::download_full_page(ConfigContainer& cfg)
+utils::curl_error RssItem::download_full_page(ConfigContainer& cfg)
 {
 	CurlHandle handle;
-	const std::string content = utils::retrieve_url(this->link(), handle, cfg, "", nullptr,
+	const auto content = utils::retrieve_url(this->link(), handle, cfg, "", nullptr,
 			utils::HTTPMethod::GET);
+	if (!content.has_value()) {
+		LOG(Level::ERROR, "RssItem::download_full_page: retrieve_url %s failed with code %d",
+			this->link(), content.error().code);
+		return utils::curl_error{content.error().code, content.error().err_msg};
```

Comment:
Wouldn't this be sufficient?

```suggestion
		return content.error().code;
```

Reply:
No, I use returned error message here
https://github.com/bogdasar1985/newsboat/blob/60e65c4a982b799131ac04f69144daf8c87fceed/src/itemviewformaction.cpp#L542